From 7de0b8455bcdf42a8d476bf45d3ceda75fe5b41c Mon Sep 17 00:00:00 2001
Message-ID: <7de0b8455bcdf42a8d476bf45d3ceda75fe5b41c.1769531391.git.3265870+tbertels@users.noreply.github.com>
From: Thomas Bertels <3265870+tbertels@users.noreply.github.com>
Date: Thu, 22 Jan 2026 12:13:37 +0100
Subject: [PATCH] settingswindow: Use Wayland mode by default on Wayland

In XWayland mode, if the mouse cursor wasn't in the window, it doesn't
get hidden after switching to fullscreen until it is moved.
This seems like a kwin_wayland bug. Other applications which run in
XWayland mode - like VLC and SMPlayer - suffer from the same problem.

The Wayland mode shouldn't have any regression with our current default
settings.
---
 src/settingswindow.ui | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/settingswindow.ui b/src/settingswindow.ui
index 6a60060..0e58245 100644
--- a/src/settingswindow.ui
+++ b/src/settingswindow.ui
@@ -7804,6 +7804,9 @@ media file played</string>
              <property name="text">
               <string>Prefer Wayland over XWayland (restart required)</string>
              </property>
+             <property name="checked">
+              <bool>true</bool>
+             </property>
             </widget>
            </item>
            <item row="7" column="0" colspan="2">
-- 
2.52.0

